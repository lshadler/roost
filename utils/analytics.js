
import ReactGA from 'react-ga';

/**
 * initialize Analytics
 */
export const initGA = () => {
  ReactGA.initialize('UA-161523112-1');
};

/**
 *
 */
export const logPageView = () => {
  ReactGA.set({page: window.location.pathname});
  ReactGA.pageview(window.location.pathname);
};

/**
 * Logs normal window event
 *
 * @param {String} category the type of log
 * @param {String} action link
 */
export const logEvent = (category = '', action = '') => {
  if (category && action) {
    ReactGA.event({category, action});
  }
};

/**
 * Logs Error
 *
 * @param {String} description details about the exception
 * @param {boolean} fatal if the page
 */
export const logException = (description = '', fatal = false) => {
  if (description) {
    ReactGA.exception({description, fatal});
  }
};
