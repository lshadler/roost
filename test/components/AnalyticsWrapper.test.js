import React from 'react';

jest.mock('../../utils/analytics', () => ({
  initGA: jest.fn(),
  logPageView: jest.fn(),
}));

import analytics from '../../utils/analytics';

import {render} from '@testing-library/react';

import AnalyticsWrapper from '../../components/AnalyticsWrapper';

describe('Analytics Wrapper Component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('initializes analytics and logs page view', () => {
    const {asFragment} = render(<AnalyticsWrapper>Test</AnalyticsWrapper>);

    expect(asFragment()).toMatchSnapshot();
    expect(analytics.initGA).toHaveBeenCalledTimes(1);
    expect(analytics.logPageView).toHaveBeenCalledTimes(1);
  });

  test('only logs page view if analytics already defined', () => {
    window.GA_INITIALIZED = true;
    const {asFragment} = render(<AnalyticsWrapper>Test</AnalyticsWrapper>);

    expect(asFragment()).toMatchSnapshot();
    expect(analytics.initGA).toHaveBeenCalledTimes(0);
    expect(analytics.logPageView).toHaveBeenCalledTimes(1);
  });
});
