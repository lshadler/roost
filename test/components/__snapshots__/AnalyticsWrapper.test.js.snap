// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics Wrapper Component initializes analytics and logs page view 1`] = `
<DocumentFragment>
  Test
</DocumentFragment>
`;

exports[`Analytics Wrapper Component only logs page view if analytics already defined 1`] = `
<DocumentFragment>
  Test
</DocumentFragment>
`;

exports[`Analytics Wrapper Component renders in default state 1`] = `
<DocumentFragment>
  Test
</DocumentFragment>
`;
